DiverseSelector/selectors.py

```diff
@@ -138,52 +178,70 @@ def optisim(self, arr):
-        if len(result) == num_select:
-            return result
-
-        low = rg if len(result) > num_select else 0
-        high = rg if low == 0 else None
-        bounds = [low, high]
-        count = 0
-        while len(result) != num_select and count < 20:
-            if bounds[1] is None:
-                rg = bounds[0] * 2
-            else:
-                rg = (bounds[0] + bounds[1]) / 2
-            self.r = rg
-            result = self.optisim(arr)
-            if len(result) > num_select:
-                bounds[0] = rg
-            else:
-                bounds[1] = rg
-            count += 1
-        self.r = None
-        return result
+    def select_from_cluster(self, arr, num_selected, cluster_ids=None):
+        """
+        Algorithm that uses optisim for selecting points from cluster.
+        Parameters
+        ----------
+        arr: np.ndarray
+            coordinate array of points
+        num_selected: int
+            number of molecules that need to be selected.
+        cluster_ids: np.ndarray
+            indices of molecules that form a cluster
+
+        Returns
+        -------
+        selected: list
+            list of ids of selected molecules
+        """
+        return predict_radius(self, arr, num_selected, cluster_ids)
 
 
 class DirectedSphereExclusion(SelectionBase):
-    def __init__(self, r=None, func_distance=lambda x, y: np.linalg.norm(x - y), random_seed=42):
+    """Selecting points using Directed Sphere Exclusion algorithm.
+
+    Starting point is chosen as the reference point and not included in the selected molecules. The distance of each
+    point is calculated to the reference point and the points are then sorted based on the ascending order of distances.
+    The points are then evaluated in their sorted order, and are selected if their distance to all the other selected
+    points is at least r away. Euclidian distance is used by default and the r value is automatically generated if
+    not passed to satisfy the number of molecules requested.
+    """
+    def __init__(self, r=None, func_distance=lambda x, y: np.linalg.norm(x - y), start_id=0, random_seed=42):
+        """
+        Initializing class.
+
+        Parameters
+        ----------
+        r: float
+            radius for directed sphere exclusion algorithm, no points within r distance to an already selected point can
+             be selected.
+        func_distance: callable
+            function for calculating the pairwise distance between instances of the array.
+        start_id: int
+            index for the first point to be selected.
+        random_seed: int
+            seed for random selection of points be evaluated.
+        """
         self.r = r
         self.random_seed = random_seed
-        self.starting_idx = 0
+        self.starting_idx = start_id
         self.func_distance = func_distance
 
-    def sphere_exclusion(self, arr):
+    def algorithm(self, arr):
```

Comment:
Probably, we can change `algorithm` to a different name? This function name will lead to ambiguity at some point. Thanks.